### PR TITLE
cephadm: add reconfig service action

### PR DESCRIPTION
**Pull request by liewegas**
- refresh ceph.conf and/or keyring without touching running container
- refresh configs inside prom container
- restart non-ceph containers/daemons

### CODE REVIEW

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -780,9 +784,9 @@ def deploy_daemon(fsid, daemon_type, daemon_id, c, uid, gid,
             fsid, daemon_type, daemon_id,
             uid, gid,
             config, keyring)
-
-    deploy_daemon_units(fsid, uid, gid, daemon_type, daemon_id, c,
-                        osd_fsid=osd_fsid)
+    if not reconfig:
+        deploy_daemon_units(fsid, uid, gid, daemon_type, daemon_id, c,
```
Comment (sebastian-philipp) [inline src/cephadm/cephadm]: Which means we're not restarting or notifying the the daemon at all here?

Comment (liewegas) [reply src/cephadm/cephadm]: Correct.  For ceph daemons that's fine.  For prom, we need to sighup or something?  @pcuzner?

**Comment by liewegas** [comment]
retest this please

File: src/pybind/mgr/orchestrator.py [hunk 2]
```diff
@@ -857,7 +857,7 @@ def service_action(self, action, service_type, service_name=None, service_id=Non
         * If using service_id, perform the action on a single specific daemon
           instance.
 
-        :param action: one of "start", "stop", "reload", "restart", "redeploy"
```
Comment (tchaikov) [inline src/pybind/mgr/orchestrator.py]: this introduced a regression. #32518 is posted to addressed it.

**Pull request merged**